application/app/views/shared/_file_stats_widget.html.erb

```diff
@@ -0,0 +1,70 @@
+<%# Badge counts (ensure values default to 0) %>
+<% completed ||= 0 %>
+<% pending ||= 0 %>
+<% cancelled ||= 0 %>
+<% error ||= 0 %>
+<% total ||= 0 %>
+
+<%
+  if total == 0
+    percent = 0
+  else
+    percent = ((completed + cancelled + error) * 100 / total.to_f).round(1).to_i
+  end
+%>
+<% offset = (114 - (114 * percent.to_f / 100)).round(1).to_i %>
+
+<%# Default colors if not passed %>
+<% progress_color ||= "#0d6efd" %>
+<% background_color ||= "#e9ecef" %>
+<% text_color ||= "#212529" %>
+
+<div class="project-progress-container d-flex flex-column align-items-center justify-content-start">
+  <div class="fw-semibold mb-2" style="font-size: 0.9rem;">
+    <%= title %>
+  </div>
+
+  <div id="progress-circle-<%= id %>"
+       class="project-progress d-flex align-items-center gap-3">
+    <!-- Circle -->
+    <div class="flex-shrink-0 h-100">
+      <svg viewBox="0 0 44 44" width="100%" height="100%" class="progress-circle">
+        <circle cx="22" cy="22" r="18"
+                stroke="<%= background_color %>"
+                fill="none"
+                stroke-width="6" />
+        <g transform="rotate(-90 22 22)">
+          <circle data-progress-circle-target="circle"
+                  cx="22" cy="22" r="18"
+                  stroke="<%= progress_color %>"
+                  fill="none"
+                  stroke-width="6"
+                  stroke-dasharray="114"
+                  stroke-dashoffset="<%= offset %>"
+                  stroke-linecap="butt"
+                  style="transition: stroke-dashoffset 0.6s ease;" />
+        </g>
+        <text x="22" y="22"
+              dy="1.5"
+              data-progress-circle-target="text"
+              font-size="0.65rem"
+              fill="<%= text_color %>"
+              dominant-baseline="middle"
+              text-anchor="middle">
+          <%= percent.to_i %>%
+        </text>
+      </svg>
+    </div>
+
+    <!-- Badges -->
+    <div class="progress-badge-stack cursor-default">
+      <span class="badge badge-project-pending" title="Number of pending files"><%= pending %></span>
+      <span class="badge badge-project-completed" title="Number of completed files"><%= completed %></span>
+      <div class="d-flex gap-1">
+        <span class="badge badge-half badge-project-cancel" title="Number of cancelled files"><%= cancelled %></span>
+        <span class="badge badge-half badge-project-error" title="Number of error files"><%= error %></span>
+      </div>
+      <span class="badge badge-project-total" title="Total files"><%= total %></span>
```

Comment:
Can you update the titles to "Pending files" instead of "Number of pending files" and so on? We updated it that way in other views in another PR